### PR TITLE
delete SYS_MC_EST_GROUP

### PR DESCRIPTION
**Pull request by dagar**
Delete SYS_MC_EST_GROUP and introduce per module control. This opens up the possibility of having an invalid config (eg both EKF2 + LPE enabled), so I added a few basic checks in LPE & ATTQ init if the relevant topics are already published.

In general having centralized config parameters like SYS_MC_EST_GROUP just doesn't work well if we can't guarantee the relevant pieces are actually in the build.

This level of control should also be helpful for simulation setups that bypass the sensors pipeline + estimator.

Needs corresponding user guide updates. https://github.com/PX4/PX4-user_guide

### CODE REVIEW

File: src/modules/attitude_estimator_q/attitude_estimator_q_params.c [hunk 1]
```diff
@@ -39,6 +39,16 @@
...
 
+/**
+ * standalone attitude estimator enable (unsupported)
+ *
+ * Enable standalone quaternion based attitude estimator.
+ *
+ * @group Attitude Q estimator
+ * @boolean
+ */
+PARAM_DEFINE_INT32(ATT_W_EN, 0);
```
Comment (bresch) [inline src/modules/attitude_estimator_q/attitude_estimator_q_params.c:50]:
```suggestion
PARAM_DEFINE_INT32(ATT_EN, 0);
```
or `ATT_Q_EN`.

**Review by bresch — DISMISSED** [review]

**Comment by qaz6906** [comment]
#. ground test
  1. Utilization of equipment such as Pixhawk 6X, VN300, Lift&Cruise type VTOL, etc.
  2. Parameter settings
    a. SENS_VN_CFG : telem2
    b. VN_MODE: INS
    c. SYS_HAS_MAG: 0
    d. EKF2_EN: Disabled
  3. Sharing ground test results
    a. Global position and local position are marked as invalid in QGC
    b. Position control mode is not authorized

1  QGC error

    c. EKF2 on PX4 is still listed as enabled

2  ekf2


data :

[3. data.zip](https://github.com/PX4/PX4-Autopilot/files/13916613/3.data.zip)

**Review by julianoes — APPROVED** [review]

**Comment by hamishwillee** [comment]
@dagar I can update if you can tell me how we now enable the different estimators: EKF2, LPE, Q.

The places to update are:
- https://docs.px4.io/main/en/advanced/switching_state_estimators.html#how-to-enable-different-estimators
- https://docs.px4.io/main/en/advanced_config/tuning_the_ecl_ekf.html#how-do-i-use-the-ecl-library-ekf 
- https://docs.px4.io/main/en/config_mc/racer_setup.html#estimator
- https://docs.px4.io/main/en/config_vtol/vtol_without_airspeed_sensor.html#optional-recommended-parameters
- https://docs.px4.io/main/en/ros/external_position_estimation.html#lpe-tuning-configuration

This showed up here https://github.com/PX4/PX4-user_guide/pull/3188#issuecomment-2060231751

**Comment by bmelanman** [comment]
Hello! 

I just to happened to catch a bug that I think was added by the above commit. When the EKF2 module is not included in the build, it seems that [this line](https://github.com/PX4/PX4-Autopilot/blob/78bbb66568e6a47291dc4d96ed177318704075fb/src/modules/commander/HealthAndArmingChecks/checks/estimatorCheck.cpp#L103) causes the message `ERROR [parameters] get: param 65535 invalid` to flood the terminal, and adding the EKF2 module to the build fixes the issue.

I made the following edits as a temporary patch:

```
int32_t param_ekf2_en = 0, param_ekf2_en_id = param_find_no_notification("EKF2_EN");

if (param_ekf2_en_id != PARAM_INVALID) {
	param_get(param_ekf2_en_id, &param_ekf2_en);
}
```

Hope this helps :)

**Comment by bresch** [comment]
@bmelanman Thanks for reporting, I made a fix here: https://github.com/PX4/PX4-Autopilot/pull/23056

**Pull request merged**